packages/cli/src/commands/pg/settings/auto-explain.ts

```diff
@@ -0,0 +1,36 @@
+import {Command, flags} from '@heroku-cli/command'
+import {Args} from '@oclif/core'
+import heredoc from 'tsheredoc'
+import {generate, boolean} from '../../../lib/pg/setter'
+// ref: https://www.postgresql.org/docs/current/auto-explain.html
+function explain(setting: {value: string}) {
+  if (setting.value) {
+    return 'Execution plans of queries will be logged for future connections.'
+  }
+
+  return 'Execution plans of queries will not be logged for future connections.'
+}
+
+export default class AutoExplain extends Command {
+  static topic = 'pg';
+  static description = heredoc(`
+  Automatically log execution plans of queries without running EXPLAIN by hand.
+  The auto_explain module is loaded at session-time so existing connections will not be logged.
+  Restart your Heroku app and/or restart existing connections for logging to start taking place.'
+  `)
+
+  static flags = {
+    app: flags.app({required: true}),
+    remote: flags.remote(),
+  }
+
+  static args = {
+    database: Args.string(),
+    value: Args.string(),
+  }
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(AutoExplain)
+    return generate('auto_explain', boolean, explain)({app: flags.app, args}, this.heroku)
```

Comment:
What do you think about this pattern with a shared `generate` function? Now with Typescript and classes, would this be better expressed via inheritance? Like, have a `PgSettingsCommand` that all these commands extend?

Reply:
Great idea. Done!